lib/std/fs/wasi.zig

```diff
@@ -173,8 +224,6 @@ test "extracting WASI preopens" {
 
     try preopens.populate();
 
-    try std.testing.expectEqual(@as(usize, 1), preopens.asSlice().len);
-    const preopen = preopens.find(PreopenType{ .Dir = "." }) orelse unreachable;
-    try std.testing.expect(preopen.@"type".eql(PreopenType{ .Dir = "." }));
-    try std.testing.expectEqual(@as(i32, 3), preopen.fd);
+    const preopen = preopens.find(PreopenType{ .Dir = "/cwd" }) orelse unreachable;
```

Comment:
So what's the difference between `.` and `/cwd`?

Reply:
It's purely a convention for the Preopens, but the primary advantage is that "/cwd" is an absolute path that supports `getcwd` and `realpath`. 

For instance, if the CWD is set to a preopen backed by "/cwd", then realpath("./some/file/path") will return "/cwd/some/file/path".

If the CWD is backed by just ".", then realpath has to error out instead.

Reply:
There's an alternative I hadn't considered: `/preopen/fd/<N>/` could be used as a special absolute path corresponding to a preopen file descriptor.

realpath("./some/file/path") would return "/preopen/fd/\<N\>/some/file/path" and `resolvePathWasi` would check for the special `/preopen/fd/` prefix when resolving absolute paths, so that it can return the correct file descriptor. I'm not sure whether that's worth implementing right now, but I'm open to feedback 

In any case, this territory is a bit uncharted - wasi-libc doesn't currently emulate getcwd() or realpath()

Reply:
I like the sound of the latter; however, this seems like a hefty amount of work (unless I am mistaken) so we leave that out for a subsequent PR. Would you agree?

Reply:
Hope it's not a bother to ask for a quick re-review  - I ended up getting this one done in [f357598](https://github.com/ziglang/zig/pull/11021/commits/f3575983ebd10b01fb9e4af427355f176ebd0ba6)